Game2048Intelligence/src/main/java/heuristic/PreferUpHeuristic.java

```diff
@@ -0,0 +1,47 @@
+package heuristic;
+
+import models.game2048.KeyEventHandler;
+import models.game2048.Scoreboard;
+import models.grid2048.Grid2048;
+import java.util.Random;
+
+public class PreferUpHeuristic extends GameHeuristic {
+    public PreferUpHeuristic(Grid2048 grid2048, Scoreboard scoreboard) {
+        super(grid2048, scoreboard);
+    }
+
+    @Override
+    public KeyEventHandler evaluateNextGameState() {
+        KeyEventHandler[] eventSequence = establishKeyEventSequence();
+        int idx = 0;
+        while (idx < 4) {
+            if (eventSequence[idx].isTilesMoved()) {
+                return eventSequence[idx];
+            }
+            idx++;
+       }
+        throw new IllegalStateException("Initial board empty or world ends");
+    }
+
+    KeyEventHandler[] establishKeyEventSequence() {
```

Comment:
Raw arrays are usually discouraged over things that inherit the `Collections` interface like List Set or Map, because the arrays are harder to work with.